plugins/resources/compound/libcompound.cpp

```diff
@@ -533,10 +534,15 @@ namespace {
 
     int close_replica(
         irods::plugin_context& _ctx,
-        const int l1descInx) {
+        const int l1descInx,
+        const int error_code_set_to_oprStatus_on_dest_close = 0) {
```

Comment:
This input also impacts the call to close for the source replica, so I think the bit about `on_dest_close` is a little misleading. I do appreciate how descriptive this is, so maybe something to indicate that this is for the operation status without the mention of source or destination? Also could consider splitting the function in two: one for each replica. Just my two cents

Reply:
I am only currently setting the oprStatus on the destination close as the source isn't being altered and I don't think the status of the source should be altered.

Reply:
I vote to pull the `oprStatus` change out of `close_replica` if possible. Pushing the handling of that value into `close_replica` doesn't appear to offer much.

Reply:
I am going to move it outside of `close_replica` but I am not sure that is a great design either.  We are introducing code that needs to be inserted every time `close_replica` is called.

Reply:
A couple of points.

1.  If I move oprStatus outside of `close_replica` I don't think it would make any sense to leave anything other than the actual call to `rsDataObjClose` inside that method.  At that point it wouldn't make a lot of sense to even have `close_replica`.

2.  I was completely wrong with my first comment on this.  For some reason I had it in my head that `close_replica` was closing both source and destination rather than it being called twice - once for each.  I am not sure why I was thinking that but I agree that my variable naming makes no sense in light of the fact that there are two calls to `close_replica`.

Reply:
I updated the variable name to `operation_status` as described above.  I did not use a leading underscore to be consistent with existing code.